File: D2.MasterData/Models/UnboundSubUnit.cs
using D2.MasterData.Parameters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace D2.MasterData.Models
{
    public class UnboundSubUnit: SubUnit
    {
        protected UnboundSubUnit() { }

        public UnboundSubUnit(UnboundSubUnitParameters argument, AdministrationUnit administrationUnit)
        {
            Id = argument.Id;
            Title = argument.Title;
            Number = argument.Number;
            Version = argument.Version;
            AdministrationUnit = administrationUnit;
            Type = argument.Type;
        }


        public virtual UnboundSubUnitType? Type
        {
            get;
            protected set;
        }
    }
}

File: D2.UI/D2.UI/src/app/masterdata/subunit/unbound-subunit.ts
import { SubUnit } from './subunit';
import { UnboundSubUnitType } from './unbound-subunit-type';

export class UnboundSubunit extends SubUnit {
  public Type?: UnboundSubUnitType;
  constructor (id: string, title: string, version: number, usage: string, number: number, type: UnboundSubUnitType) {
    super();
    this.Id = id;
    this.Title = title;
    this.Version = version;
    this.Number = number;
    this.Type = type;
  }
}
